controller/internal/enforcer/nfqdatapath/afinetrawsocket/afinetrawsocket_windows.go

```diff
@@ -59,7 +59,7 @@ func (w *PacketMetadata) udpForward(buf []byte, version packet.IPver) error {
 		w.PacketInfo.Ipv4 = 0
 	}
 	w.PacketInfo.PacketSize = uint32(len(buf))
-	r1, _, err1 := frontman.PacketFilterForwardProc.Call(uintptr(unsafe.Pointer(&w.PacketInfo)), uintptr(unsafe.Pointer(&buf[0])))
+	r1, err1 := frontman.Driver.PacketFilterForward(uintptr(unsafe.Pointer(&w.PacketInfo)), uintptr(unsafe.Pointer(&buf[0])))
```

Comment:
This doesn't just apply to this file, but I would have changed the Driver APIs to accept the real types and then in the wrapper function have it do the uintptr(unsafe.Pointer(...)) calls.  That way if an API is called in multiple places, the caller doesn't have to do this, and it make the code cleaner in my opinion.

Reply:
That might be a good improvement, but my goal here was to just create an interface for mocking. If instead it's a kind of wrapper that converts and massages types for the DLL, then we're not testing that type-conversion code. That would be my argument.

Reply:
I understand you are trying to create a mocking layer, but your mock is only testing your mock.  It doesn't test the driver.  If a type conversion is wrong, then you really won't know it until you call the driver.  Also your mock isn't testing if the type conversions are correct or I didn't see this any where.  Maybe I missed it.  Also, if you look at the internals of the windows syscalls, their APIs don't have you call them with uintptr as parameters.  They have a clean interface where the real types are passed.

Reply:
Right, I'm not testing the type conversions. I am testing the rest of the Windows Go code that calls into the driver, by presenting a "dumb mock" of the driver so that the unit test can execute without needing a driver running. I get your suggestion that: providing a wrapper for this code would be nice, and testing the type conversion would be nice too.

Reply:
Actually let me correct that a little. The controller/internal/supervisor/iptablesctrl/iptables_windows_test.go implements the mock and does some basic type conversions that we expect the driver to make.

Reply:
So to me, having everyone doing the same type conversion also increases the possibility of doing it wrong and having others copy and paste the same mistake from some where else.  So if you just did the type conversion in one place, then you still can test it, and it would be less error prone.  For example, let say in your wrapper function.  Instead of putting uintptr(unsafe.Pointer(...)) every where.  You could create functions like convertPacketToNativeType or convertStringToNativeType.  Then you can test these functions and separately as well and would improve readability as well.

Reply:
I can see the argument for that, and I'll work on that. The duplication of uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(name))) type of code can be improved.